libntxm/arm7/source/player.cpp

```diff
@@ -559,11 +574,28 @@ void Player::handleEffects(void)
 							}
 							break;
 						}
+						
+						case(EFFECT_E_PATTERN_DELAY):
+						{
+							if (effstate.pattern_delay == 0)
+							{
+								effstate.pattern_delay_store = e_effect_param + 1;
+							}
+							break;
+						}
 					}
 
 					break;
 				}
 
+				case EFFECT_POSITION_JUMP:
+				{
+					effstate.pattern_break_requested = true;
+					effstate.pattern_break_row = 0;
+					state.potpos = param-1;
```

Comment:
This does appear to be fine on paper, but for future maintenance I would rather see a new effect state flag (like `pattern_break_requested`) and doing the change in `calcNextPos` like the other effects do.

Reply:
I think adding an additional flag will make the implementation of the feature less accurate / more convoluted, as pattern break does effectively cause a position jump, the position for which can be overridden by Dxx commands only to the right of the pattern break.

But I could rename some things and move the position change itself into `calcNextPos` as requested, if that sounds good?

Reply:
It could be the same flag, that's an implementation detail. What I mean is that the position should be changed only once the entire row has been processed.

(Unless XM players *do* actually jump mid-processing and take further channels from the jumped-to pattern? But that would be strange.)

Reply:
Made the change - adding a second flag turned to be the cleanest way to do it after all.

Reply:
Thank you!